wp-parsely.php

```diff
@@ -47,12 +226,44 @@ function maybe_load_plugin() {
 	 *
 	 * add_filter( 'wpvip_parsely_load_mu', '__return_false' );
 	 */
-	if ( ! apply_filters( 'wpvip_parsely_load_mu', get_option( '_wpvip_parsely_mu' ) === '1' ) ) {
-		return;
-	}
+	$do_not_load_plugin = ! apply_filters( 'wpvip_parsely_load_mu', get_option( '_wpvip_parsely_mu' ) === '1' );
+
+	// Check if wp-parsely has already been loaded, for example when the user
+	// has a self-managed wp-parsely installation.
+	$plugin_is_already_loaded = in_array( PARSELY_PLUGIN_SIGNATURE, get_option( 'active_plugins' ) );
+
+	// Check if wp-parsely is in a state of attempting to be activated
+	// for a self-managed wp-parsely installation.
+	$plugin_loading_queued = is_queued_for_activation();
+
+	// If the plugin must not be loaded, set Loader Info data and exit.
+	if ( $do_not_load_plugin || $plugin_is_already_loaded || $plugin_loading_queued ) {
```

Comment:
This is ... incredibly complicated with not enough comments. It also feels like this is in the wrong order. It would benefit from very descriptive variable names and early returns, like:

```
if ( $self_managed_plugin_has_loaded || $self_managed_plugin_is_queued_for_activation ) {
  Parsely_Loader_Info::set_active( true );
  Parsely_Loader_Info::set_integration_type( Parsely_Loader_Info::INTEGRATION_TYPE_SELF_MANAGED );
  return;
}

if ( $load_plugin_requested_via_option && $force_disabled ) {
  Parsely_Loader_Info::set_active( false );
  Parsely_Loader_Info::set_integration_type( Parsely_Loader_Info::INTEGRATION_TYPE_MU_PLUGINS );
  return;
}

// etc
```

Reply:
Thank you for the excellent suggestions for this very convoluted logic. This section has been one of the most difficult things about working on this piece of code. Just understanding what's happening is a nightmare. Thanks to your suggestion I have made this section a lot less convoluted and more readable, and I've added a lot of relevant comments to help the next person who might have to read this. I feel that the latest commit makes this section a lot better. Thank you very much for your time and great thoughts on this.